cognite/client/_cognite_client.py

```diff
@@ -213,3 +217,93 @@ def default_oauth_interactive(
         """
         credentials = OAuthInteractive.default_for_azure_ad(tenant_id, client_id, cdf_cluster)
         return cls.default(project, cdf_cluster, credentials, client_name)
+
+    @classmethod
+    def load(cls, config: dict) -> CogniteClient:
+        """Loads a dictionary of configuration fields into a cognite client object.
+
+        Args:
+            config (dict): A dictionary containing configuration values needed to create a CogniteClient.
+
+        Returns:
+            CogniteClient: A cognite client object.
+
+        Examples:
+                Create a cognite client object from a dictionary input:
+
+                >>> from cognite.client import CogniteClient
+                >>> import os
+                >>> config = {
+                ...     "client_name": "abcd",
+                ...     "project": "cdf-project",
+                ...     "base_url": "https://api.cognitedata.com/",
+                ...     "client_credentials": {
+                ...         "client_id": "abcd",
+                ...         "client_secret": os.environ["OAUTH_CLIENT_SECRET"],
+                ...         "token_url": os.environ["TOKEN_URL"],
+                ...         "scopes": ["https://greenfield.cognitedata.com/.default"],
+                ...         # Any additional IDP-specific token args. e.g.
+                ...         "audience": "some-audience",
+                ...     }
+                ... }
+                >>> client = CogniteClient.load(config)
+        """
+        return cls(ClientConfig.load(config))
+
+    @classmethod  # TODO: design discussion on if we should have this method or not, and if it should sub envs or not
+    def from_yaml(cls, file_path: str | Path) -> CogniteClient:
+        """Loads a YAML file containing configuration fields into a cognite client object.
+        Any environment variables in the YAML file will be replaced with their defined values given they are referenced
+        using the following syntax: ${ENV_VAR_NAME} (recommended) or $ENV_VAR_NAME.
+
+        Note: The environment variables must be defined in the current environment and there are no implicit environment
+                variables available in the YAML file (e.g. CDF_PROJECT will not automatically replace the project name
+                unless `project: ${CDF_PROJECT}` is defined in the YAML file).
+
+        Args:
+            file_path (str | Path): The path to the YAML file containing the configuration values needed to create a CogniteClient.
+
+        Returns:
+            CogniteClient: A cognite client object.
+
+        Examples:
+                Create a cognite client object from a YAML file, using envs from the current environment:
+
+                >>> config.yaml
+                >>> project: $MY_CDF_PROJECT
+                >>> base_url: https://${MY_CDF_CLUSTER}.cognitedata.com/
+                >>> client_credentials:
+                >>>     token: ${MY_CDF_TOKEN}
```

Comment:
All cognite resources assume camelCase in the load method. Suggest following the same practice here

Reply:
These are all instances of user input, not responses from the CogniteAPI, so I would think the snake case is the standard to use @haakonvt?

Reply:
I discussed this with @nodegard and camelCase makes a lot of sense when loading resources from the API (which is 99.95% of what we do in the SDK), but for these configuration/setup classes, just matching parameters 1:1 (i.e. snake_case), makes the most sense - to us at least.